util/util.go

```diff
@@ -405,3 +406,63 @@ func Remount(d *schema.ResourceData, client *api.Client, mountField string, isAu
 
 	return ret, nil
 }
+
+type RetryRequestOpts struct {
+	MaxTries    uint64
+	Delay       time.Duration
+	StatusCodes []int
+}
+
+func (r *RetryRequestOpts) IsRetryableStatus(statusCode int) bool {
+	for _, s := range r.StatusCodes {
+		if s == statusCode {
+			return true
+		}
+	}
+
+	return false
+}
+
+func DefaultRequestOpts() *RetryRequestOpts {
+	return &RetryRequestOpts{
+		MaxTries:    60,
+		Delay:       time.Millisecond * 500,
+		StatusCodes: []int{http.StatusBadRequest},
+	}
+}
+
+// RetryWrite attempts to retry a Logical.Write() to Vault for the
+// RetryRequestOpts. Primary useful for handling some of Vault's eventually
+// consistent APIs.
+func RetryWrite(client *api.Client, path string, data map[string]interface{}, req *RetryRequestOpts) (*api.Secret, error) {
```

Comment:
Nice! Are we now also good to remove `writeSecreteDataWithRetry` from [this line](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_kv_secret_v2.go#L394) since it would no longer be used?

Reply:
Ah, good catch! FIxed with a4ced5baf23b58c45d2547ba9b71610b2e78a140